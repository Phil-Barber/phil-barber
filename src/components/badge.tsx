import React from 'react';
import styled from 'styled-components';

const StyledBadge = styled.button<{ readonly isSelected: boolean }>`
  border: none;
  outline: none;
  cursor: pointer;
  font-size: 12px;
  display: inline-block;
  width: fit-content;
  padding: 0 ${({ theme }) => theme.spacing.xSmall};
  border-radius: ${({ theme }) => theme.spacing.large};
  margin-right: ${({ theme }) => theme.spacing.xxSmall};

  ${({ isSelected, theme }) =>
    isSelected
      ? `
    background-color: ${theme.colors.darkGrey1};
    color: ${theme.colors.lightGrey1};
  `
      : `
    background-color: ${theme.colors.lightGrey1};
    color: ${theme.colors.darkGrey1};
  `}
`;

interface Props {
  children: React.ReactNode;
  className?: string;
  isSelected?: boolean;
  onClick?: () => void;
}

export const Badge: React.FC<Props> = ({
  children,
  className,
  isSelected = false,
  onClick,
}: Props) => (
  <StyledBadge className={className} isSelected={isSelected} onClick={onClick}>
    {children}
  </StyledBadge>
);
