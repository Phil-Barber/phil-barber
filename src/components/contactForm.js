import React from 'react'
import { useFormik } from 'formik'
import * as Yup from 'yup'


export const ContactForm = () => {
  const bodyLimit = 300

  const formik = useFormik({
    initialValues: {
      email: '',
      name: '',
      body: '',
    },
    validationSchema: Yup.object({
      email: Yup.string()
        .email('Invalid email address')
        .required('Required'),
      name: Yup.string()
        .required('Required'),
      body: Yup.string()
        .max(bodyLimit, `Must be ${bodyLimit} characters or less`)
        .required('Required'),
    }),
    onSubmit: values => {
      alert(JSON.stringify(values, null, 2));
    },
  });
  return (
    <form onSubmit={formik.handleSubmit}>
      <label htmlFor="email">Email Address</label>
      <input
        name="email"
        {...formik.getFieldProps('email')}
      />
      {formik.touched.email && formik.errors.email && <div>{formik.errors.email}</div>}
      <label htmlFor="name">Name</label>
      <input
        name='name'
        {...formik.getFieldProps('name')}
      />
      {formik.touched.name && formik.errors.name && <div>{formik.errors.name}</div>}
      <label htmlFor="body"></label>
      <input
        name='body'
        {...formik.getFieldProps('body')}
      />
      {formik.touched.body && formik.errors.body && <div>{formik.errors.body}</div>}
      <button type="submit">Submit</button>
    </form>
  );
};


