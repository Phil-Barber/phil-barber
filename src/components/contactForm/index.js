import React from 'react';
import { Formik, Form } from 'formik';
import * as Yup from 'yup';
import * as S from './index.styled';

const URL =
  'https://8hot7pcq6e.execute-api.eu-west-2.amazonaws.com/dev/email/send';

export const ContactForm = () => {
  const bodyLimit = 300;

  return (
    <Formik
      initialValues={{
        email: '',
        name: '',
        body: '',
      }}
      validationSchema={Yup.object({
        email: Yup.string()
          .email('Invalid email address')
          .required('Required'),
        name: Yup.string().required('Required'),
        company: Yup.string(),
        body: Yup.string()
          .max(bodyLimit, `Must be ${bodyLimit} characters or less`)
          .required('Required'),
      })}
      onSubmit={(values, { setSubmitting, resetForm, setStatus }) => {
        const req = new XMLHttpRequest();
        req.open('POST', URL, true);
        req.setRequestHeader('Content-Type', 'application/json');
        req.addEventListener('load', function() {
          setSubmitting(false);
          if (req.status < 400) {
            // Success
            resetForm({});
            setStatus({ success: true });
          } else {
            setStatus({ success: false });
            throw Error('Request failed: ' + req.statusText);
          }
        });
        req.send(JSON.stringify(values));
      }}
    >
      {formik => (
        <>
          <Form>
            <S.FullWidthInput
              label="Email Address*"
              name="email"
              type="email"
            />
            <S.FlexBox>
              <S.ShortInput label="Name*" name="name" type="text" />
              <S.ShortInput label="Company" name="company" type="text" />
            </S.FlexBox>
            <S.MessageInput
              label="Message*"
              name="body"
              textarea
              placeholder={`Hi Phil,

Hope you're having a great day!

Listen to this really exciting project you could contribute to:`}
            />
            <S.FlexBox>
              <S.SubmitButton type="submit" disabled={formik.isSubmitting}>
                Submit
              </S.SubmitButton>
              <S.RequiredText>*Required field</S.RequiredText>
            </S.FlexBox>
          </Form>
        </>
      )}
    </Formik>
  );
};
