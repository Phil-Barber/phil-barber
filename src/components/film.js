import React from 'react';
import Layout from './layout';

const Film = ({ children, filmTitle, filmPoster }) => (
  <Layout>
    <h1>{filmTitle}</h1>
    {children}
    <img src={filmPoster} alt={`${filmTitle} Poster`} />
  </Layout>
);

export default Film;
