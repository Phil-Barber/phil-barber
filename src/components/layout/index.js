import React from 'react';
import styled from 'styled-components';
import Headroom from 'react-headroom';
import { css } from '@emotion/core';
import { useStaticQuery, Link, graphql } from 'gatsby';

import { rhythm } from '../../utils/typography';

const MainHeader = styled.div`
  background-color: ${({ theme }) => theme.colors.primary};
  padding: ${({ theme }) => `${theme.spacing.xxSmall} ${theme.spacing.normal}`};

  h1 {
    color: ${({ theme }) => theme.colors.secondary};
    margin-bottom: 0;
  }
`;

const SecondaryHeader = styled.div`
  background-color: ${({ theme }) => theme.colors.secondary};
  color: ${({ theme }) => theme.colors.primary};
  border-bottom: 2px solid ${({ theme }) => theme.colors.tertiary};
  padding: ${({ theme }) => `
    ${theme.spacing.xxSmall} 
    ${theme.spacing.large}
    ${theme.spacing.xxxSmall}
  `};
  display: flex;
  justify-content: flex-end;
`;

const StyledLink = styled(Link)`
  text-decoration: none;
`;

const NavLink = styled(StyledLink)`
  font-size: 18px;
  color: ${({ theme }) => theme.colors.primary};
`;

const ContentContainer = styled.div`
  background-color: ${({ theme }) => theme.colors.tertiary};
  color: ${({ theme }) => theme.colors.secondary};
`;

const Content = styled.div`
  margin: 0 auto;
  max-width: 750px;
  padding-top: ${({ theme }) => theme.spacing.large};
  text-align: justify;
`;

const Layout = ({ children }) => {
  const data = useStaticQuery(
    graphql`
      query {
        site {
          siteMetadata {
            title
          }
        }
      }
    `
  );
  return (
    <>
      <Headroom>
        <MainHeader>
          <StyledLink to={'/'}>
            <h1
              css={css`
                margin-bottom: ${rhythm(2)};
                font-style: normal;
              `}
            >
              {data.site.siteMetadata.title}
            </h1>
          </StyledLink>
        </MainHeader>
        <SecondaryHeader>
          <NavLink to={'/contact/'}>
            <div>Contact</div>
          </NavLink>
        </SecondaryHeader>
      </Headroom>
      <ContentContainer>
        <Content>{children}</Content>
      </ContentContainer>
    </>
  );
};

export default Layout;
