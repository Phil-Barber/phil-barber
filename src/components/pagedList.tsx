import React, { useState } from 'react';
import styled from 'styled-components';

const PageButtons = styled.button`
  margin: 0px ${({ theme }) => theme.spacing.xxSmall};
`;

interface Props {
  children: React.ReactNode[];
  pageSize?: number;
}

export const PagedList: React.FC<Props> = ({
  children,
  pageSize = 10,
}: Props) => {
  const numPages = Math.ceil(children.length / pageSize);
  const [pageNum, setPageNum] = useState<number>(0);

  const start = pageSize * pageNum;
  return (
    <>
      {children.slice(start, start + pageSize)}
      <PageButtons
        disabled={pageNum === 0}
        onClick={() => setPageNum((num) => num - 1)}
      >
        Previous
      </PageButtons>
      {Array.from({ length: numPages }, (_, n) => (
        <PageButtons
          disabled={pageNum === n}
          key={n}
          onClick={() => setPageNum(n)}
        >
          {n + 1}
        </PageButtons>
      ))}
      <PageButtons
        disabled={pageNum === numPages - 1}
        onClick={() => setPageNum((num) => num + 1)}
      >
        Next
      </PageButtons>
    </>
  );
};
