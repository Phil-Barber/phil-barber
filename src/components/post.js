import React from 'react';
import styled from 'styled-components';
import { Link } from 'gatsby';
import { useHover } from '../hooks/useHover';

const PostContainer = styled.div`
  padding: ${({ theme }) => theme.spacing.xxSmall};
  padding-bottom: 0;
  transition: background-color 1s ease-out;
  text-align: left;
  :hover {
    background-color: white;
  }
`;

const Excerpt = styled.p`
  visibility: hidden;
  max-height: 0px;
  overflow: hidden;
  transition: max-height 1s, visibility 1s;
  position: relative;

  ${({ isHovered }) =>
    isHovered &&
    `
    max-height: 100px;
    visibility: visible;
  `};

  :after {
    content: '';
    width: 100%;
    height: 100%;
    position: absolute;
    left: 0;
    top: 0;
    background: linear-gradient(
      transparent 40px,
      ${({ theme }) => theme.colors.white}
    );
  }
`;

const StyledLink = styled(Link)`
  text-decoration: none;
  color: inherit;
`;

const Date = styled.span`
  color: #667269;
`;

export const Post = ({ fields, frontmatter, excerpt }) => {
  const [hoverRef, isHovered] = useHover();

  return (
    <div ref={hoverRef}>
      <PostContainer>
        <StyledLink to={fields.slug}>
          <h3>
            {frontmatter.title}
            <Date> â€” {frontmatter.dateCompleted}</Date>
          </h3>
          <Excerpt isHovered={isHovered}>{excerpt}</Excerpt>
        </StyledLink>
      </PostContainer>
    </div>
  );
};
