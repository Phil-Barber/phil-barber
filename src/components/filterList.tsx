import React, { useState } from 'react';
import { Badge } from '../components/badge';

type items = React.ReactNode[];
type filters = { [key: string]: (lookup: any) => boolean };
type lookupFunc = (item: any) => string;
interface Props {
  items: items;
  filters: filters;
  lookupFunc: lookupFunc;
  children: (filtered: React.ReactNode[]) => React.ReactNode;
}

const useFilters = (
  items: items,
  filters: filters,
  lookupFunc: lookupFunc
): [Array<string>, React.ReactNode[], (clicked: string) => () => void] => {
  const [selected, setSelected] = useState<Array<string>>([]);
  const toggleSelected = (clicked: string) => () => {
    setSelected((selected) =>
      selected.includes(clicked)
        ? selected.filter((item) => item !== clicked)
        : [...selected, clicked]
    );
  };

  const filteredItems = items.filter((item) => {
    if (!selected.length) return true;

    return selected.reduce(
      (acc, key) => acc || filters[key](lookupFunc(item)),
      false
    );
  });

  return [selected, filteredItems, toggleSelected];
};

export const FilterList: React.FC<Props> = ({
  items,
  filters,
  lookupFunc,
  children,
}: Props) => {
  const [selected, filteredItems, toggleSelected] = useFilters(
    items,
    filters,
    lookupFunc
  );

  const filterButtons = Object.keys(filters).map((key) => (
    <Badge
      key={key}
      onClick={toggleSelected(key)}
      isSelected={selected.includes(key)}
    >
      {key}
    </Badge>
  ));

  return (
    <>
      {filterButtons}
      {children(filteredItems)}
    </>
  );
};
