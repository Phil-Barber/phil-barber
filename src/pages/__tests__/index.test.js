import React from 'react';
import { render } from '../../../testUtils';
import { Main } from '../index';
import * as post from '../../components/post';

describe('Main', () => {
  const mockPost = () => <div>Mock Post</div>;
  post.AnimatedPost = mockPost;

  it('Matches snapshot', () => {
    const { container } = render(
      <Main
        data={{
          markdownRemark: {
            html: '<div>Test About Content',
          },
          allMarkdownRemark: {
            totalCount: 2,
            edges: [{ node: { id: 1 } }, { node: { id: 2 } }],
          },
        }}
      />
    );
    expect(container).toMatchSnapshot();
  });
});
