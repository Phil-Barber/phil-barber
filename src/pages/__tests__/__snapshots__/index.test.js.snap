// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main Matches snapshot 1`] = `
<div>
  <div
    class="headroom-wrapper"
  >
    <div
      class="headroom headroom--unfixed"
      style="position: relative; top: 0px; left: 0px; right: 0px; z-index: 1; transform: translate3D(0, 0, 0);"
    >
      <div
        class="sc-AxjAm fEzfgO"
      >
        <a
          class="sc-AxhUy bILkkK"
          href="/"
        >
          <h1>
            Test title
          </h1>
        </a>
        <span
          class="sc-AxirZ gtVPNy"
          style="transform: rotate(0deg);"
        >
          &gt;
        </span>
      </div>
    </div>
  </div>
  <div
    class="sc-AxiKw jmqPVm"
  >
    <div
      class="sc-AxheI gPjnvY"
    >
      <div
        class="sc-fznKkj hZbvix"
      >
        <div
          class="sc-fzplWN sc-fznZeY fLOyIP"
        >
          <div
            class="sc-fzokOt cIzEVk"
          >
            <div>
              Test About Content
            </div>
          </div>
        </div>
        <div
          class="sc-fzplWN sc-fznyAO gTZffj"
        >
          <div
            class="sc-fzqBZW fNVHmc"
          >
            <h1>
              Posts
            </h1>
            <span
              class="sc-fzqNJr TZybT"
            >
              Total: 
              2
            </span>
          </div>
          <div>
            Mock Post
          </div>
          <div>
            Mock Post
          </div>
        </div>
      </div>
    </div>
    <div
      class="sc-AxhCb jDuQaB"
    />
  </div>
</div>
`;
